### PR TITLE
Reference the setup-earthly github actions

### PR DESCRIPTION
**Pull request by alexcb**
Update the GHA example to include an example of using the
earthly/actions/setup-earthly action.

Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

**Comment by alexcb** [comment]
we have [77 users of this now](https://github.com/earthly/actions-setup/network/dependents); and have received a few PRs to improve it.

I think it's fair to say we've gone through a few bags of dogfood :yum:

**Comment by alexcb** [comment]
(if merged, i'll cherry-pick this into `docs-0.7`)

File: docs/ci-integration/guides/gh-actions-integration.md [hunk 1]
```diff
@@ -1,7 +1,52 @@
 
 # GitHub Actions integration
 
-Here is an example a GitHub Actions build, where we build the Earthly target `+build`.
+Here is an example a GitHub Actions build that uses the [https://github.com/earthly/actions/setup-earthly](earthly/actions/setup-earthly action).
```
Comment (nelsam) [inline docs/ci-integration/guides/gh-actions-integration.md:4]:
```suggestion
Here is an example of a GitHub Actions build that uses the [https://github.com/earthly/actions/setup-earthly](earthly/actions/setup-earthly action).
```

File: docs/ci-integration/guides/gh-actions-integration.md [hunk 2]
```diff
@@ -1,7 +1,52 @@
...
+        git checkout -b "$branch" || true
+    - name: Docker Login
+      run: docker login --username "$DOCKERHUB_USERNAME" --password "$DOCKERHUB_TOKEN"
+    - name: Earthly version
+      run: earthly --version
+    - name: Run build
+      run: earthly --push +build
+```
+
+Alternatively, you can skip using the `earthly/actions/setup-earthly` job, and include
```
Comment (nelsam) [inline docs/ci-integration/guides/gh-actions-integration.md:48]: I think this sentence flows slightly better with one less comma, but that's purely an opinion.
```suggestion
Alternatively, you can skip using the `earthly/actions/setup-earthly` job and include
```

**Review by nelsam — APPROVED** [review]

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**